### PR TITLE
[example][tracing] Add simple example for `promptflow-tracing`

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

This PR targets to add a simple example that can work with `promptflow-tracing`: implement a simple OTLP collector, and user can view the collected traces from stdout with simple configuration.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: examples/tracing/otlp-collector/llm.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import os
+
```
Comment (wangchao1230) [inline examples/tracing/otlp-collector/llm.py:2]: maybe add this to this PR https://github.com/microsoft/promptflow/pull/1741

We can we merge them together

File: examples/tracing/otlp-collector/otlp-collector.ipynb [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from llm import my_llm_tool\n",
+    "\n",
+    "deployment_name = \"text-davinci-003\""
```
Comment (wangchao1230) [inline examples/tracing/otlp-collector/otlp-collector.ipynb:89]: new a new model as text-davinci-003 is deprecated?

File: examples/tracing/otlp-collector/.env.example [hunk 3]
```diff
@@ -0,0 +1,3 @@
+AZURE_OPENAI_API_KEY=<your_AOAI_key>
+AZURE_OPENAI_API_BASE=<your_AOAI_endpoint>
```
Comment (wangchao1230) [inline examples/tracing/otlp-collector/.env.example:2]: Let's put the .env.example in a parent folder, as they can share across examples. 
See this one
https://github.com/microsoft/promptflow/pull/1741/files#diff-b2cfa42efc087d918fe05319dbd27aebbcb516300de250f83a4d7e5508f9c318

File: examples/tracing/otlp-collector/otlp-collector.ipynb [hunk 4]
```diff
@@ -0,0 +1,135 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
```
Comment (wangchao1230) [inline examples/tracing/otlp-collector/otlp-collector.ipynb:5]: add some description at the head of the notebook describe the purpose and what user can gain from this

File: examples/tracing/otlp-collector/otlp-collector.ipynb [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "result = my_llm_tool(\n",
+    "    prompt=\"Write a simple Hello, world! program that displays the greeting message when executed.\",\n",
+    "    deployment_name=deployment_name,\n",
+    ")\n",
+    "result\n",
+    "# view the traces under this cell"
```
Comment (wangchao1230) [inline examples/tracing/otlp-collector/otlp-collector.ipynb:103]: why the log will be printed in this cell?

File: examples/tracing/otlp-collector/otlp-collector.ipynb [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "from promptflow.tracing import start_trace\n",
+    "\n",
+    "os.environ[\"PF_TRACING_SKIP_LOCAL_SETUP\"] = \"true\"\n",
```
Comment (wangchao1230) [inline examples/tracing/otlp-collector/otlp-collector.ipynb:62]: It's not very intuitive to expose environ here. Maybe something like

start_trace(exporter=otlp_span_exporter)

Comment (zhengfeiwang) [reply examples/tracing/otlp-collector/otlp-collector.ipynb:62]: this is a workaround before we really separate `promptflow-tracing`, the real purpose here is to simulate user does not install devkit deps; but I think you are right, we might need some ways like above.

**Comment by zhengfeiwang** [comment]
will move this example to #1741 .

**Pull request closed without merging**